File: src/main/java/no/ks/fiks/virksomhetsertifikat/VirksomhetSertifikatAutoConfigure.java
package no.ks.fiks.virksomhetsertifikat;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties({KsVirksomhetSertifikaterProperties.class})
public class VirksomhetSertifikatAutoConfigure {

    @Bean
    public KsVirksomhetSertifikater virksomhetSertifikater(KsVirksomhetSertifikaterProperties props){
        return new KsVirksomhetSertifikater(props);
    }
}

File: virksomhetssertifikat/src/main/kotlin/no/ks/fiks/virksomhetsertifikat/VirksomhetSertifikatAutoConfigure.kt
package no.ks.fiks.virksomhetsertifikat

import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class VirksomhetSertifikatAutoConfigure {
    @ConditionalOnMissingBean(VirksomhetSertifikater::class)
    @Configuration
    @EnableConfigurationProperties(
        VirksomhetSertifikaterProperties::class
    )
    internal class AutomatiskVirksomhetSertifikat {
        @Bean
        fun virksomhetSertifikater(props: VirksomhetSertifikaterProperties): VirksomhetSertifikater {
            return VirksomhetSertifikater(props.virksomhetssertifikater)
        }
    }
}